### PR TITLE
Implement customized path and crosspoints

### PR DESCRIPTION
**Pull request by zhoney**
implemented: #305

Change-Id: I4b21fd0ccc287eed08e2494c04731a7cd909ea8d

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    private long limit;
+    @JsonProperty("with_path")
+    private boolean withPath;
+    @JsonProperty("with_vertex")
+    private boolean withVertex;
+
+    private CrosspointsRequest() {
+        this.sources = null;
+        this.pathPatterns = new ArrayList<>();
+        this.capacity = 10000000;
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java]: 10_000_000L, 10_000L

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    @JsonProperty("limit")
+    private long limit;
+    @JsonProperty("with_vertex")
+    private boolean withVertex;
+
+    private PathsRequest() {
+        this.sources = null;
+        this.steps = new ArrayList<>();
+        this.sortBy = SortBy.NONE;
+        this.capacity = 10000000;
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java]: ditto

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java (hunk 1 above)
Comment (zhoney) [reply src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java]: done

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    private long limit;
+    @JsonProperty("with_path")
+    private boolean withPath;
+    @JsonProperty("with_vertex")
+    private boolean withVertex;
+
+    private CrosspointsRequest() {
+        this.sources = null;
+        this.pathPatterns = new ArrayList<>();
+        this.capacity = 10_000_000L;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/structure/CrosspointsRequest.java]: use Traverser.DEFAULT_XX

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    @JsonProperty("limit")
+    private long limit;
+    @JsonProperty("with_vertex")
+    private boolean withVertex;
+
+    private PathsRequest() {
+        this.sources = null;
+        this.steps = new ArrayList<>();
+        this.sortBy = SortBy.NONE;
+        this.capacity = 10_000_000L;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java]: ditto

File: src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        @JsonProperty("sample")
+        private long sample;
+
+        private Step() {
+            this.direction = null;
+            this.labels = new ArrayList<>();
+            this.properties = new HashMap<>();
+            this.weightBy = null;
+            this.defaultWeight = 0.0D;
+            this.degree = 10_000L;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/structure/PathsRequest.java]: ditto

File: src/main/java/com/baidu/hugegraph/structure/constant/Traverser.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.structure.constant;
+
+public class Traverser {
+
+    public static final String DEFAULT_CAPACITY = "10000000";
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/constant/Traverser.java]: prefer to set long

**Pull request merged**